agent/consul/acl_client.go

```diff
@@ -23,23 +23,28 @@ var clientACLCacheConfig *structs.ACLCachesConfig = &structs.ACLCachesConfig{
 	Roles: 128,
 }
 
-func (c *Client) ACLDatacenter() string {
-	// For resolution running on clients, servers within the current datacenter
+type clientACLResolverBackend struct {
+	// TODO: un-embed
```

Comment:
Could the client and server be un-embedded in this PR?

Reply:
Good question! I had another look to see if that was possible. It looks like it's not because the `Client` and `Server` are still being used directly for the `RPC()` method, and also for `EnterpriseACLResolverDelegate`.

I've got another TODO about splitting up the RPC into more specific calls (ex: `ReadToken`, `PolicyResolve`). I think by doing that we could have the `Server` implementation call the endpoint directly, without going through the in-memory codec that uses reflection.

But that's definitly a larger change, and one we'll need to think about more, as it could impact which metrics are emitted for these calls.  So I think that's probably better left for a future PR.